power_grid_model_io_native_c/power_grid_model_io_native/include/power_grid_model_io_native/pgm_vnf_converter/converter_parser.hpp

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_IO_NATIVE_C_CONVERTER_PARSER_HPP
+#define POWER_GRID_MODEL_IO_NATIVE_C_CONVERTER_PARSER_HPP
+
+#include <power_grid_model_io_native/common/common.hpp>
+
+#include <power_grid_model/auxiliary/input.hpp>
+#include <power_grid_model/container.hpp>
+
+#include <iostream>
+#include <map>
+#include <regex>
+
+namespace power_grid_model_io_native {
+using VisionGUID = std::string;
+
+struct VnfNode {
+    VisionGUID guid{"nan"};
+    double u_nom{nan};
+};
+
+using VnfGrid = pgm::Container<VnfNode>;
+template <typename Identifier> using IdentifierLookup = std::map<Identifier, ID>;
+using VisionGUIDLookup = IdentifierLookup<VisionGUID>;
+
+class PgmVnfParser {
+  public:
+    explicit PgmVnfParser(std::string_view vnf_data);
```

Comment:
no `explicit` here; it prevents e.g. `std::string` or `char const*` from being converted to `std::string_view` (which is a zero-copy pass-by-value view of the original data and therefore equally or better in terms of efficiency and memory safety than `std::string const&`)